rocky/reports/views/base.py

```diff
@@ -197,6 +210,51 @@ def get_report_type_ids(self) -> list[str]:
     def get_report_types(self) -> list[type[BaseReport]]:
         return [get_report_by_id(report_type_id) for report_type_id in self.selected_report_types]
 
+    @staticmethod
+    def get_report_types_from_ooi_selelection(
+        report_types: set[type[BaseReport]] | set[type[Report]] | set[type[MultiOrganizationReport]],
+    ) -> list[dict[str, str]]:
+        """
+        The report types are fetched from which ooi is selected. Shows all report types for the oois.
+        """
+
+        return [
+            {
+                "id": report_type.id,
+                "name": report_type.name,
+                "description": report_type.description,
+                "label_style": report_type.label_style,
+            }
+            for report_type in report_types
+        ]
+
+    def get_report_types_for_generate_report(self):
+        return self.get_report_types_from_ooi_selelection(get_report_types_for_oois(self.selected_oois))
+
+    def get_report_types_for_aggregate_report(self) -> dict[str, list[dict[str, str]]]:
+        reports_dict = get_report_types_from_aggregate_report(AggregateOrganisationReport)
+        report_types: dict[str, list[dict[str, str]]] = {}
+
+        for option, reports in reports_dict.items():
+            report_types[option] = self.get_report_types_from_ooi_selelection(reports)
+        return report_types
+
+    def get_available_report_types(self) -> tuple[list[dict[str, str]] | dict[str, list[dict[str, str]]], int]:
+        report_types: list[dict[str, str]] | dict[str, list[dict[str, str]]] = {}
+        if self.report_type is not None:
```

Comment:
You don't need to check for `None` here if you're not doing anything with it and especially if you're already testing against other conditions